ikarus/finiteelements/mechanics/materials/neohooke.hh

```diff
@@ -72,11 +72,16 @@ struct NeoHookeT : public Material<NeoHookeT<ST>>
    * \return ScalarType The stored energy.
    */
   template <typename Derived>
-  ScalarType storedEnergyImpl(const Eigen::MatrixBase<Derived>& C) const noexcept {
+  ScalarType storedEnergyImpl(const Eigen::MatrixBase<Derived>& C) const {
     static_assert(Concepts::EigenMatrixOrVoigtNotation3<Derived>);
     if constexpr (!Concepts::EigenVector<Derived>) {
-      const auto traceC  = C.trace();
-      const auto logdetF = log(sqrt(C.determinant()));
+      const auto traceC = C.trace();
+      const auto detC   = C.determinant();
+      if (Dune::FloatCmp::lt(static_cast<double>(detC), 0.0, 1e-10))
+        DUNE_THROW(Dune::InvalidStateException,
+                   "Determinant of right Cauchy Green tensor C must be greater than or equal to zero. detC = " +
```

Comment:
It should be equal to zero. It should really be bigger. From theory.

Reply:
You mean it shouldn't be equal to zero? So change `FloatCmp::lt` to `FloatCmp::le`?

Reply:
Ah sorry. I missed a word. It should **not** be equal to zero. Therefore `detC>0`. The statement in the if statement is fine but the exception text is saying `detC>=0`.

Reply:
I fixed it